ext/curl/interface.c

```diff
@@ -1634,7 +1634,10 @@ static bool php_curl_set_callable_handler(zend_fcall_info_cache *const handler_f
 #define HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER(curl_ptr, constant_no_function, handler_type) \
 	case constant_no_function##FUNCTION: { \
 		bool result = php_curl_set_callable_handler(&curl_ptr->handlers.handler_type->fcc, zvalue, is_array_config, #constant_no_function "FUNCTION"); \
-		if (!result) { \
+		if (!result \
+		    || (curl_ptr->handlers.read && !curl_ptr->handlers.read->fcc.function_handler) \
+		    || (curl_ptr->handlers.write && !curl_ptr->handlers.write->fcc.function_handler) \
+		    || (curl_ptr->handlers.write_header && !curl_ptr->handlers.write_header->fcc.function_handler)) { \
 			return FAILURE; \
 		} \
 		curl_ptr->handlers.handler_type->method = PHP_CURL_USER; \
```

Comment:
As mentioned, I believe the `method` needs to be reset when the handler is null. I.e.

https://github.com/php/php-src/blob/6e172f0ac19d308dac3aeaf0485de34f57d9a644/ext/curl/interface.c#L1239-L1241